peer/brontide.go

```diff
@@ -1537,6 +1537,18 @@ out:
 			case <-p.quit:
 				break out
 			}
+
+			// Send the Shutdown to the link. If ProcessCloseMsg
+			// later fails validating the Shutdown and this message
+			// is sent to the link, the coop close process won't
+			// happen but the link will be unaware. This is fine as
+			// the link should eventually send a Shutdown and stop
+			// when the channel has no htlc's or updates. It will
+			// notify us that coop close is ready, and gracefully
+			// exit.
+			targetChan = msg.ChannelID
+			isLinkUpdate = p.isActiveChannel(msg.ChannelID)
```

Comment:
Use `p.tryLinkShutdown` instead to avoid re-implementing the same logic?

Reply:
Not sure what you mean here?

Reply:
was thinking we could just remove the link here so we don't need to re-implement the same set of logic in link again, but I see this method is now removed so nvm.